### ISSUE TITLE
False positive: indentation inside of constructor parameters inside try block

### ISSUE THREAD

**Issue by lrozenblyum** [checkstyle/checkstyle#18614]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```cmd
E:\temp\CheckstyleReport>type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="tabWidth" value="4"/>

  <module name="TreeWalker">
    <module name="Indentation" />
  </module>
</module>

E:\temp\CheckstyleReport>type Test.java
public class Test {
        private Test client;

        private Test(String string) {
        }

        private void test() {
                try {
                        client = new Test(
                                null // unexpected violation about indentation
                        );
                } catch (Exception e) {
                }
        }
}
E:\temp\CheckstyleReport>set RUN_LOCALE="-Duser.language=en -Duser.country=US"

E:\temp\CheckstyleReport>java %RUN_LOCALE% -jar checkstyle-13.0.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] E:\temp\CheckstyleReport\Test.java:10:17: 'new' child has incorrect indentation level 16, expected level should be 20. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**
The violation shouldn't be reported: the parameter passed to the constructor is already indented correctly.

NOTE: if we remove the try/catch block the violation disappears.
NOTE2: the file is formatted by tabs

---

**Comment by Rafiaminhaj** [comment]
Hi maintainers 👋  
Thanks for the detailed report and reproducer.

I can reproduce the issue locally: the Indentation check reports a violation for constructor arguments inside a try block, although the indentation looks correct.

If this issue is still open, I’d like to investigate it further and propose a fix.  
Please let me know if there is any preferred approach or related code area I should start with.

Thanks!

**Comment by romani** [comment]
issue is approved, @Rafiaminhaj , please send pull request.

**Comment by MiniPiku** [comment]
@romani i think this issue is resolved 
i tried to reproduce it but the issue passed the indentation check

i think this issue was of  `checkstyle-13.0.0-all.jar` and we are currently running  `checkstyle-13.3.0-all.jar`

**Comment by romani** [comment]
Please share CLI output same as author did to prove that it is resolved

**Comment by MiniPiku** [comment]
passing the 4 indentation check (previously failing)
<img width="2420" height="705" alt="Image" src="https://github.com/user-attachments/assets/ad5f3a1a-c4bd-463a-aa08-85ebe691e15b" />

usual indentation
<img width="2434" height="674" alt="Image" src="https://github.com/user-attachments/assets/607e9703-694f-4880-aa10-6e2b7110cf5d" />

**Comment by MiniPiku** [comment]
I can write a test for this and submit a PR referencing this issue.

**Comment by romani** [comment]
Let's do test, it would speaks by itself .


It would be awesome to share text CLI out put, same as author did

**Comment by MiniPiku** [comment]
CLI output:

```
Starting audit...
Audit done.
```

No violations were reported.

It appears this issue was already fixed in newer versions.
I will add a regression test to ensure it does not reappear.

**Comment by romani** [comment]
@MiniPiku, thanks a lot. With such test we will not regress on this

**Comment by vivek-0509** [comment]
@romani 
Please reopen the issue it is not resolved yet, the issue is specifically for tab indented files
the regression test added in PR #18960 uses spaces, not tabs, so it passes without any code fix and does not actually validate the reported scenario.
```
vivek@Viveks-MacBook-Air checkstyle % cat > /tmp/config.xml << 'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="tabWidth" value="4"/>
  <module name="TreeWalker">
    <module name="Indentation" />
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > /tmp/Test.java << 'EOF'
public class Test {
        private Test client;

        private Test(String string) {
        }

        private void test() {
                try {
                        client = new Test(
                                null
                        );
                } catch (Exception e) {
                }
        }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java -jar target/checkstyle-13.3.0-SNAPSHOT-all.jar -c /tmp/config.xml /tmp/Test.java
Starting audit...
[ERROR] /tmp/Test.java:10:17: 'new' child has incorrect indentation level 16, expected level should be 20. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by vivek-0509** [comment]
@romani 
Case with space indented, violation is correctly reported
```

vivek@Viveks-MacBook-Air checkstyle % cat > /tmp/TestWrongIndentSpaces.java << 'EOF'
public class TestWrongIndentSpaces {
    private TestWrongIndentSpaces client;

    private TestWrongIndentSpaces(String string) {
    }

    private void test() {
        try {
            client =
    new TestWrongIndentSpaces(null);
        } catch (Exception e) {
        }
    }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java -jar target/checkstyle-13.3.0-SNAPSHOT-all.jar -c /tmp/config.xml /tmp/TestWrongIndentSpaces.java

Starting audit...
[ERROR] /tmp/TestWrongIndentSpaces.java:10:5: 'new' has incorrect indentation level 4, expected level should be 16. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
Case with tab indented, got no violation, but the new is incorrectly indented
```

vivek@Viveks-MacBook-Air checkstyle % cat > /tmp/TestWrongIndent.java << 'EOF'
public class TestWrongIndent {
        private TestWrongIndent client;

        private TestWrongIndent(String string) {
        }

        private void test() {
                try {
                        client =
        new TestWrongIndent(null);
                } catch (Exception e) {
                }
        }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java -jar target/checkstyle-13.3.0-SNAPSHOT-all.jar -c /tmp/config.xml /tmp/TestWrongIndent.java

Starting audit...
Audit done.
```

the logic for tab-indented files is broken, check simply does not validate the new keyword indentation for tab-indented files due to this buggy condition, `if (getLineStart(mainAst) == mainAst.getColumnNo()) ` getLineStart() returns visual columns (tab-expanded), getColumnNo() returns character position. For tab-indented files these never match, so the if branch is dead code for them. The else branch returns new's own position as the expected indent, so the check ends up comparing new's actual position against itself, no violation can ever be reported, regardless of how wrong the indentation is.

I think the old fix in the PR https://github.com/checkstyle/checkstyle/pull/18686 to use similar condition as what `isOnStartOfLine(mainAst) `  do was the right approach. I will review the old diff report to see if there are any false negatives or false positives.

**Comment by romani** [comment]
@MiniPiku, please investigate this issue one more time, in case you will find way to resolve this issue, tab based indentation.

**Comment by MiniPiku** [comment]
@vivek-0509 thanks for mentioning the issue
@romani i believe this issue will be resolved with this PR  #19062

**Comment by MiniPiku** [comment]
@romani, I’ve identified a potential issue in the indentation test suite. The `IndentationTrailingCommentsVerticalAlignmentTest` produces false failures when input files contain tabs, reporting mismatched trailing comment alignment (e.g., expected: 77 but was: 71). This happens even though the files are intentionally crafted with tabs to trigger indentation violations, and reformatting them (tabs → spaces) defeats the purpose of the test.
the reason why i wasn't able to add a regression test in #19062 

I’m opening a new issue to document this fragility and propose separating test inputs or refining the alignment rule. Please approve so we can track and address it properly.

**Comment by romani** [comment]
fixed by https://github.com/checkstyle/checkstyle/pull/20342

**Comment by romani** [comment]
more tests are merged at https://github.com/checkstyle/checkstyle/pull/19684
